packages/backend/src/models/GroupsModel.ts

```diff
@@ -0,0 +1,164 @@
+        >(
+            `
+            INSERT INTO groups (name, organization_id)
+            SELECT ?, organization_id
+            FROM organizations
+            WHERE organization_uuid = ?
+            RETURNING group_uuid, groups.created_at
+        `,
+            [group.name, group.organizationUuid],
+        );
+        if (row === undefined) {
+            throw new UnexpectedDatabaseError(`Failed to create organization`);
+        }
+        return {
+            uuid: row.group_uuid,
+            name: group.name,
+            createdAt: row.created_at,
+            organizationUuid: group.organizationUuid,
+        };
+    }
+
+    async getGroup(groupUuid: string): Promise<Group> {
+        const [group] = await this.database('groups')
+            .innerJoin(
+                'organizations',
+                'groups.organization_id',
+                'organizations.organization_id',
+            )
+            .where('group_uuid', groupUuid)
+            .select();
+        if (group === undefined) {
+            throw new NotFoundError(`No group found`);
+        }
+        return {
+            uuid: group.group_uuid,
+            name: group.name,
+            createdAt: group.created_at,
+            organizationUuid: group.organization_uuid,
+        };
+    }
+
+    async getGroupWithMembers(groupUuid: string): Promise<GroupWithMembers> {
+        const rows = await this.database('groups')
+            .innerJoin(
+                'group_memberships',
+                'groups.group_uuid',
+                'group_memberships.group_uuid',
+            )
+            .innerJoin('users', 'group_memberships.user_id', 'users.user_id')
+            .innerJoin('emails', 'users.user_id', 'emails.user_id')
+            .innerJoin(
+                'organizations',
+                'groups.organization_id',
+                'organizations.organization_id',
+            )
+            .where('groups.group_uuid', groupUuid)
+            .andWhere('emails.is_primary', true)
+            .select();
+        if (rows.length === 0) {
+            throw new NotFoundError(`No group found`);
+        }
+        return {
+            uuid: rows[0].group_uuid,
+            name: rows[0].name,
+            createdAt: rows[0].created_at,
+            organizationUuid: rows[0].organization_uuid,
+            members: rows.map((row) => ({
+                userUuid: row.user_uuid,
+                email: row.email,
+                firstName: row.first_name,
+                lastName: row.last_name,
+            })),
+        };
+    }
+
+    async deleteGroup(groupUuid: string): Promise<void> {
+        await this.database('groups').where('group_uuid', groupUuid).del();
+    }
+
+    // TODO: do we need to enforce that only org_members from the same org can join or do we want to manage this as business logic?
+    // the current schema enforces that only org_members can be added to groups in the same org
```

Comment:
I like the schema enforces the org membership and is deleted on cascade. 
This is going to be important when we have users in multiple orgs.